addon/lib/headless.js

```diff
@@ -0,0 +1,85 @@
+/* Headless server, for use with Pageshot:
+ *  https://github.com/mozilla-services/pageshot
+ * Relies on the server module from mozilla testing:
+ *  http://mxr.mozilla.org/mozilla-central/source/netwerk/test/httpserver/
+ *
+ * Listening IP address may be assigned to httpd, by default it is localhost. 
+ * The default is used in this implementation.
+ *    For more info:
+ *    http://mxr.mozilla.org/mozilla-central/source/netwerk/test/httpserver/nsIHttpServer.idl#282
+ *
+ * TODO:
+ *    Documentation
+ *    Check URL before opening
+ *    Fix autoShot - currently it leaves some errors about attrs
+ *    Add preference for whether or not headless should run
+ *    Reimplement contentServerString?
+ * 
+ * General usage: 
+ * When you make an HTTP request to this server on the appropriate port it will
+ * look for a query string and attempt to open the string as a URL, return a
+ * URL of where to find it when processing is done, and process the tab with
+ * an automatic shot function.
+ *
+ * Generally CSS parsing is enabled on this branch, which is EXPERIMENTAL.
+ *
+ * Syntax, assuming default IP, PORT, and target URL of asdf.com:
+ *    curl 'http://localhost:10082/?http://www.asdf.com' 
+ *
+ * Usage without X:
+ * To use this without X, start the server as normal. Then, in the bin/
+ * directory, run `sh ephemeral-x.sh ./run-addon`.
+ */
+
+const { Cc, Ci, Cu } = require("chrome");
+const tabs = require("sdk/tabs");
+
+const HTTPD_MOD_URL = module.uri.replace("/headless.js", "/httpd.jsm");
```

Comment:
The user `rpl` on IRC was the one who provided that, as well as the babel patch. I completely forgot to mention that anywhere in my code, however, so i'll give him some credit here. :D